packages/syntax-core/src/WordConfetti/WordConfetti.tsx

```diff
@@ -104,15 +110,25 @@ const WordConfetti = forwardRef<HTMLDivElement, WordConfettiProps>(
       words,
     } = props;
 
+    const hashCode = useCallback((string: string) => {
```

Comment:
@jliotta should we re-write this to use `useMemo` instead? With `useCallback` we cache a function definition to ensure that the same function object is used across multiple renders. However with `useMemo` (what I think we want) we cache the return value of a function to avoid recalculating it on every render.

Reply:
@jliotta also, in this case, I would put the `hash` function outside of `WordConfetti`. Just makes it easier to ready through the code.

```tsx
function hash(input: string): number {
  //...
}

function WordConfetti() {
	const hashCode = useMemo(() => hash(word), [word]);
  return (
    //
  )
}
```

(didn't actually test but should get the idea across)

Reply:
Should I be individually memoizing the hash of each word? I memoize the entire result of `styledWords` so I think that should be sufficient

Reply:
Yes, should be sufficient